x/concentrated-liquidity/pool.go

```diff
@@ -101,6 +101,38 @@ func (k Keeper) setPool(ctx sdk.Context, pool types.ConcentratedPoolExtension) e
 	return nil
 }
 
+func (k Keeper) GetPoolDenoms(ctx sdk.Context, poolId uint64) ([]string, error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return nil, err
+	}
+
+	denoms := []string{concentratedPool.GetToken0(), concentratedPool.GetToken1()}
+	return denoms, nil
+}
+
+func (k Keeper) CalculateSpotPrice(
```

Comment:
Let's have table-driven test cases for this function, please. It is important to cover all edge cases here, including the rounding logic when we do `price = sdk.OneDec().Quo(price)` as `price` can become zero.

It would be great to separate the spot price function into a separate PR for faster merge and review cycle as well as to be able to review all edge cases in isolation.

Some of the edge cases around spot price erros can be referenced from gamm:
https://github.com/osmosis-labs/osmosis/blob/be95710eca5d4feb982c946b13716896a4e0f8f7/x/gamm/keeper/pool_service.go#L47-L53

Reply:
@p0mvn  Which case are you talking about when you are referring by the following statement? 
> the rounding logic when we do price = sdk.OneDec().Quo(price) as price can become zero.

I personally think that if the sqrt price is too big and `price` results in 0 due to 1/ <large_number> we should still be returning 0., which is how the current logic handles. Do you have any other suggestion for this?

Reply:
I could definitely split this logic out into a separate PR for faster iterations and easier reviews once the question above has been answered

Reply:
I think we should standardize the API between x/gamm and x/concentrated-liquidity. Given that x/gamm returns `sdk.Dec{}` in that case, I think we should do the same here for compatibility reasons. 

Also, a price cannot be zero, so I think returning `sdk.Dec{}` is more correct.

What's most important is returning error when spot price is zero because twap directly depends on an error being returned in such a case [here](https://github.com/osmosis-labs/osmosis/blob/106ac2058f126997e7f73e549390f065b7b02b49/x/twap/logic.go#L53)